demo/lib/vm.bash

```diff
@@ -455,25 +455,144 @@ vm-install-cri-resmgr() {
         local bin_change
         local src_change
         bin_change=$(stat --format "%Z" "$BIN_DIR/cri-resmgr")
-        src_change=$(find "$HOST_PROJECT_DIR" -name '*.go' -type f | xargs stat --format "%Z" | sort -n | tail -n 1)
+        src_change=$(find "$HOST_PROJECT_DIR" -name '*.go' -type f -print0 | xargs -0 stat --format "%Z" | sort -n | tail -n 1)
         if [[ "$src_change" > "$bin_change" ]]; then
             echo "WARNING:"
             echo "WARNING: Source files changed - installing possibly outdated binaries from"
             echo "WARNING: $BIN_DIR/"
             echo "WARNING:"
             sleep "${warning_delay:-0}"
         fi
-        host-command "$SCP \"$BIN_DIR/cri-resmgr\" \"$BIN_DIR/cri-resmgr-agent\" $VM_SSH_USER@$VM_IP:" || {
-            command-error "copying local cri-resmgr to VM failed"
-        }
-        vm-command "mv cri-resmgr cri-resmgr-agent $prefix/bin" || {
-            command-error "installing cri-resmgr to $prefix/bin failed"
-        }
+        vm-put-file "$BIN_DIR/cri-resmgr" "$prefix/bin/cri-resmgr"
+        vm-put-file "$BIN_DIR/cri-resmgr-agent" "$prefix/bin/cri-resmgr-agent"
     else
         error "vm-install-cri-resmgr: unknown binsrc=\"$binsrc\""
     fi
 }
 
+vm-cri-import-image() {
+    local image_name="$1"
+    local image_tar="$2"
+    case "$VM_CRI" in
+        containerd)
+            vm-command "ctr -n k8s.io images import \"$image_tar\" >/dev/null 2>&1; crictl --runtime-endpoint unix:///run/containerd/containerd.sock images | grep \"$image_name\"" ||
```

Comment:
Stupid question: aren't we able to examine the return value of `ctr` directly?

Reply:
Good question. This hack forced the test to run on Ubuntu 20.04 despite error from `ctr`. Anyway, that was a bad idea: the import error exit status _is_ a good reason to bail out even though the image would appear in the image list after failed importing. (This is what happens with containerd versions < 4.0.)

I verified that second importing causes an error even in the case that the same image appears in the list (after the first import having failed but image being added to images list). Therefore we actually should rely on ctr exit status rather than image list contents.

Thanks, fixed!